logwatch that retries getting docker logs if for some reason it fails

Tofel opened the pull request.

Tofel commented on logwatch/logwatch.go:
```diff
@@ -172,22 +212,117 @@ func (m *LogWatch) ConnectContainer(ctx context.Context, container testcontainer
+						Err(err).
+						Str("Container name", name).
+						Msg("Log producer errored")
+					if shouldRetry() {
```
I didn't know it existed 😅

Tofel replied:
I did use it in one place, but didn't replace all usages, because in some I want to pass incremental timeout, not just execute with the same value again and `avast/retry` doesn't support such a thing

Tofel commented on logwatch/logwatch_test.go:
```diff
@@ -221,3 +223,373 @@ func TestLogWatchDocker(t *testing.T) {
+	})
+}
+
+type MockedLogProducingContainer struct {
```
Does it support mocks having inner logic? Because here it's a must

kalverra replied:
I think so, but check their page. Core team uses it extensively, can also ask them.

Tofel replied:
I think it's a good option, but not for these tests as I need to add an extra method to that mock, that is not in the interface, for sending logs and very specific logic for starting log producer.

Tofel commented on utils/retries/retry_functions.go:
removed

Tofel commented:
Okay, added a bit of changes:
* removed pattern matching per conversations with @skudasov 
* removed saving all logs in memory per conversations with @skudasov (although you could achieve similar behaviour using `in-memory` log target)
* fixed how I was ending log producer retrying loop (done channel was defined on `LogWatch`, but it should be defined on each `consumer`, so that each container has an independent loop)
* added two more tests

skudasov commented on logstream/logstream.go:
```diff
@@ -0,0 +1,827 @@
+	}
+}
+
+// Stop stops the consumer and closes temp file
```
Maybe we change the comment, I don't see any file closing.

skudasov commented on logstream/logstream_handlers.go:
```diff
@@ -0,0 +1,234 @@
+		return errors.Wrap(err, "failed to open log file. File logging stopped")
+	}
+
+	defer logFile.Close()
```
We are closing and opening a file for each line, can we open the file once?
